olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -240,42 +246,53 @@ void AuthenticationClient::Impl::SetNetworkProxySettings(
 
 void AuthenticationClient::Impl::SetNetwork(
     std::shared_ptr<http::Network> network) {
-  if (network_ != network) {
-    network_ = std::move(network);
+  if (settings_.network_request_handler != network) {
+    settings_.network_request_handler = std::move(network);
   }
 }
 
 void AuthenticationClient::Impl::SetTaskScheduler(
     std::shared_ptr<TaskScheduler> task_scheduler) {
-  if (task_scheduler_ != task_scheduler) {
-    task_scheduler_ = std::move(task_scheduler);
+  if (settings_.task_scheduler != task_scheduler) {
+    settings_.task_scheduler = std::move(task_scheduler);
   }
 }
 
 AuthenticationClient::Impl::Impl(const std::string& authentication_server_url,
                                  size_t token_cache_limit)
-    : server_url_(authentication_server_url),
-      client_token_cache_(std::make_shared<
-                          Atomic<utils::LruCache<std::string, SignInResult> > >(
+    : client_token_cache_(
+          std::make_shared<Atomic<utils::LruCache<std::string, SignInResult>>>(
+              token_cache_limit)),
+      user_token_cache_(std::make_shared<
+                        Atomic<utils::LruCache<std::string, SignInUserResult>>>(
```

Comment:
Can you add some aliases for this two type:
```cpp
using SignInCacheType = Atomic<utils::LruCache<std::string, SignInResult>>;
using SignInUserCacheType = Atomic<utils::LruCache<std::string, SignInUserResult>>;
```

This should come in very handy and will reduce all this breaking